libs/model/src/user/GetUserReferrals.query.ts

```diff
@@ -0,0 +1,51 @@
+import { type Query } from '@hicommonwealth/core';
+import * as schemas from '@hicommonwealth/schemas';
+import { z } from 'zod';
+import { models } from '../database';
+
+export function GetUserReferrals(): Query<typeof schemas.GetUserReferrals> {
+  return {
+    ...schemas.GetUserReferrals,
+    auth: [],
+    secure: true,
+    body: async ({ actor, payload }) => {
+      // only super admin can get referrals for all users
+      const id =
+        actor.user.isAdmin && payload.user_id ? payload.user_id : actor.user.id;
+
+      const referrals = await models.Referral.findAll({
+        where: { referrer_id: id },
+        include: [
+          {
+            model: models.User,
+            as: 'referrer',
+            attributes: ['id', 'profile'],
+          },
+          {
+            model: models.User,
+            as: 'referee',
+            attributes: ['id', 'profile'],
+          },
+        ],
+      });
+
+      // format view
+      return referrals.map(
```

Comment:
Not a huge deal but usually we avoid mapping to create the response structure on the javascript side, since it is sync and will slow down subsequent requests due to the single threaded nature of javascript. Usually we write raw SQL for these kinds of things and build the json objects with `json_build_object` (see GetCommunities.query, GetMembers.query, GetThreads.query, ...)

Anyways not a huge deal since it is only an optimization issue and should not block this PR from getting in.